pkg/GAPJulia/JuliaInterface/julia/gap.jl

```diff
@@ -68,3 +68,89 @@ for (left, right) in typecombinations
         end
     end
 end
+
+
+"""
+    @gap <obj>
+    @gap(<obj>)
+
+Executes <obj> directly in GAP, as if `GAP.EvalString("<obj>")` was called.
+Can be used for creating GAP literals directly from Julia.
+
+    julia> @gap (1,2,3)
+    GAP: (1,2,3)
+    julia> @gap SymmetricGroup(3)
+    GAP: SymmetricGroup( [ 1 .. 3 ] )
+    julia> @gap(SymmetricGroup)(3)
+    GAP: SymmetricGroup( [ 1 .. 3 ] )
+
+Note that the last two examples have a slight syntactical, and therefore also
+a semantical difference. The first one executes the string `SymmetricGroup(3)`
+directly inside GAP. The second example returns the function `SymmetricGroup`
+via `@gap(SymmetricGroup)`, then calls that function with the argument `3`.
+"""
+macro gap(str)
+    return EvalString(string(str))
+end
+
+export @gap
+
+"""
+    LoadPackageAndExposeGlobals(package::String, mod::String; all_globals::Bool = false)
+    LoadPackageAndExposeGlobals(package::String, mod::Module = Main; all_globals::Bool = false)
+
+`LoadPackageAndExposeGlobals` loads `package` into GAP via `LoadPackage`,
+and stores all newly defined GAP globals as globals in the module `mod`. If `mod` is
+a string, the function creates a new module, if `mod` is a Module, it uses `mod` directly.
+
+The function is intended to be used for creating mock modules for GAP packages.
+If you load the package `CAP` via
+
+    LoadPackageAndExposeGlobals( "CAP", "CAP" )
+
+you can use CAP commands via
+
+    CAP.PreCompose( a, b )
+
+"""
+function LoadPackageAndExposeGlobals(package::String, mod::String; all_globals::Bool = false)
+    mod_sym = Symbol(mod)
+    Base.MainInclude.eval(:(
+        module $(mod_sym)
+            import GAP
+        end
+    ))
+    ## Adds the new module to the Main module, so it is directly accessible in the julia REPL
+    mod_mod = Base.MainInclude.eval(:(Main.$(mod_sym)))
+
+    ## We need to call `invokelatest` as the module `mod_mod` was only created during the
+    ## call of this function in a different module, so its world age is higher than the
+    ## function calls world age.
+    Base.invokelatest(LoadPackageAndExposeGlobals, package, mod_mod; all_globals = all_globals)
+end
+
+function LoadPackageAndExposeGlobals(package::String, mod::Module; all_globals::Bool = false)
+    current_gvar_list = EvalString("ShallowCopy(NamesGVars())")
```

Comment:
Minor remark: We only need this if `all_globals` is false. So we may want to execute it conditionally, no? Also, couldn't you write `Globals.ShallowCopy(Globals.NamesGVars())` instead of using `EvalString`?